test/extended/images/mongodb_replica.go

```diff
@@ -0,0 +1,58 @@
+package images
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/db"
+)
+
+var _ = g.Describe("[images][mongodb] openshift mongodb replication", func() {
+	defer g.GinkgoRecover()
+
+	const (
+		templatePath     = "https://raw.githubusercontent.com/openshift/mongodb/master/2.4/examples/replica/mongodb-clustered.json"
+		expectedReplicas = 3 // FIXME: get it from the template
+	)
+
+	oc := exutil.NewCLI("mongodb-replica", exutil.KubeConfigPath()).Verbose()
+
+	g.Describe("creating from a template", func() {
+		g.It(fmt.Sprintf("should process and create the %q template", templatePath), func() {
+
+			g.By("creating a new app")
+			o.Expect(oc.Run("new-app").Args("-f", templatePath).Execute()).Should(o.Succeed())
+
+			g.By("waiting for the deployment to complete")
+			exutil.WaitForADeployment(
+				oc.KubeREST().ReplicationControllers(oc.Namespace()),
+				"mongodb",
+				exutil.CheckDeploymentCompletedFn,
+				exutil.CheckDeploymentFailedFn,
+			)
+
+			g.By(fmt.Sprintf("expecting that there are %v running mongodb pods", expectedReplicas))
+			podNames, err := exutil.WaitForPods(
+				oc.KubeREST().Pods(oc.Namespace()),
+				exutil.ParseLabelsOrDie("name=mongodb-replica"),
+				exutil.CheckPodIsRunningFn,
+				expectedReplicas,
+				1*time.Minute,
+			)
+			o.Expect(err).ShouldNot(o.HaveOccurred())
+			o.Expect(podNames).Should(o.HaveLen(expectedReplicas))
+
+			g.By(fmt.Sprintf("expecting that replica set have %v members", expectedReplicas))
+			mongo := db.NewMongoDB(podNames[0], "")
+			getReplicaHostsCmd := "printjson(db.isMaster().hosts.length)"
+			result, err := mongo.Query(oc, getReplicaHostsCmd)
+			o.Expect(err).ShouldNot(o.HaveOccurred())
+			o.Expect(result).Should(o.Equal(string(expectedReplicas)))
```

Comment:
This is not how to convert int to string.
You can use `strconv.Itoa(expectedReplicas)`. Assuming `result` is a string, because normally I'd expect it to be a `[]byte` coming from a `exec.Cmd`...

https://golang.org/pkg/strconv/#Itoa